src/index.jsx

```diff
@@ -0,0 +1,17 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import {Provider} from "react-redux"
+import {createStore, applyMiddleware} from "redux"
+import thunk from "redux-thunk"
+import rootReducer from "./reducer"
+import * as action_creators from "./action_creators"
```

Comment:
agreed, you wouldn't pass down from top level, you would have each container component be aware of its own action creators - so in this case your App component is your stateful 'container', so should be the one that imports action creators - also, get that underscore bullshit out of JS - :camel: